sdk/web-pubsub/web-pubsub-client/src/webPubSubClient.ts

```diff
@@ -356,10 +380,14 @@ export class WebPubSubClient {
     eventName: string,
     content: JSONTypes | ArrayBuffer,
     dataType: WebPubSubDataType,
-    options?: SendEventOptions
-  ): Promise<WebPubSubResult> {
+    options?: SendEventOptionsInternal
+  ): Promise<void | WebPubSubResult> {
```

Comment:
Why would this be void or some value?  Is this breaking expected?  It never returns a value so shouldn't they all be `void`?

Reply:
It depends on whether it's fire-and-forget. If it is, then nothing comes from the service, otherwise, service will return a response. As it's an internal function. For public, it's a overload. For fireandforget on, the signature shows returning `promise<void>`

Reply:
We should split out these functions that have different return values so that the customer isn't confuse why there is no return value in some cases and other cases not such as `sendEvent` and `sendEventFireAndForget` or something along those lines.

Reply:
As we discussed in the review meeting. We can have the overloads for this method. If user passed in `{ fireAndForget: true }`, then it returns `Promise<void>` and if not, it will return the value from service which will be `Promise<WebPubSubResult>`. That can be clear in IntelliSense. And the difference of return come from the service.

Reply:
Ok we can close for now then